### DISCUSSION TITLE
Organizing output class list into sections manually

### DISCUSSION THREAD

**Discussion by balajivenkatesh** [lsegal/yard#1573] [Q&A] [answered]
[Edit - simplified to one question since they were similar
Question:
We have classes with same module names in `app/models` and `app/lib`.
Example:
app/model/vehicle/car.rb, `class Vehicle::Car`
app/model/vehicle/truck.rb, `class Vehicle::Truck`
app/lib/vehicle/driver/assign.rb, `class Vehicle::Driver::Assign`
We also have a module in app/model/vehicle.rb, `module Vehicle`

YARD groups the models under one item in the output class list. But not the lib files.

Is there a way to fix this with config or folder/file structures?

![Screen Shot 2020-10-13 at 12 58 12 AM](https://user-images.githubusercontent.com/4346757/95832525-3886af80-0cef-11eb-812a-b3caeead050b.png)

**Comment by lsegal** [comment, marked as answer]
YARD doesn't group based on directory or know about "models" as a concept. The reason you're seeing disconnected trees is because YARD has no context for the "Driver" code object in your hierarchy. 

Based on layout you've described, there is no file that defines `Driver` as either a module or a class, and because it's undefined in your source, YARD treats it as an external class. As an external class, it is not generated as an expandable field.

Defining `Vehicle::Driver` as a module somewhere-- anywhere-- in your source should resolve this. Avoiding the shorthand class reopening syntax (see Rubocop's [Style/ClassAndModuleChildren](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/ClassAndModuleChildren) rule) would be an easy way to do this. In `lib/vehicle/driver/assign.rb` do this:

```ruby
module Vehicle
  module Driver
    class Assign
      # code here
    end
  end
end
```

Now you will get the correct tree:

![image](https://user-images.githubusercontent.com/686/97096410-b9fc0d00-1620-11eb-968e-a652dd06eb95.png)
